The job **`x86_64-gnu-llvm-18`** failed! Check out the build log: [(web)](https://github.com/rust-lang/rust/actions/runs/12011228399/job/33479787350) [(plain)](https://github.com/rust-lang/rust/commit/c66da836a612710a6a7b9f208803599600d41ff6/checks/33479787350/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
#16 exporting to docker image format
#16 sending tarball 28.1s done
#16 DONE 34.3s
##[endgroup]
Setting extra environment values for docker:  --env ENABLE_GCC_CODEGEN=1 --env GCC_EXEC_PREFIX=/usr/lib/gcc/
[CI_JOB_NAME=x86_64-gnu-llvm-18]
debug: `DISABLE_CI_RUSTC_IF_INCOMPATIBLE` configured.
---
sccache: Starting the server...
##[group]Configure the build
configure: processing command line
configure: 
configure: build.configure-args := ['--build=x86_64-unknown-linux-gnu', '--llvm-root=/usr/lib/llvm-18', '--enable-llvm-link-shared', '--set', 'rust.randomize-layout=true', '--set', 'rust.thin-lto-import-instr-limit=10', '--enable-verbose-configure', '--enable-sccache', '--disable-manage-submodules', '--enable-locked-deps', '--enable-cargo-native-static', '--set', 'rust.codegen-units-std=1', '--set', 'dist.compression-profile=balanced', '--dist-compression-formats=xz', '--set', 'rust.lld=false', '--disable-dist-src', '--release-channel=nightly', '--enable-debug-assertions', '--enable-overflow-checks', '--enable-llvm-assertions', '--set', 'rust.verify-llvm-ir', '--set', 'rust.codegen-backends=llvm,cranelift,gcc', '--set', 'llvm.static-libstdcpp', '--enable-new-symbol-mangling']
configure: target.x86_64-unknown-linux-gnu.llvm-config := /usr/lib/llvm-18/bin/llvm-config
configure: llvm.link-shared     := True
configure: rust.randomize-layout := True
configure: rust.thin-lto-import-instr-limit := 10
---
  Downloaded boml v0.3.1
   Compiling boml v0.3.1
   Compiling y v0.1.0 (/checkout/compiler/rustc_codegen_gcc/build_system)
    Finished `release` profile [optimized] target(s) in 3.95s
     Running `/checkout/obj/build/x86_64-unknown-linux-gnu/stage1-codegen/x86_64-unknown-linux-gnu/release/y test --use-system-gcc --use-backend gcc --out-dir /checkout/obj/build/x86_64-unknown-linux-gnu/stage1-tools/cg_gcc --release --mini-tests --std-tests`
Using system GCC
[BUILD] example
[AOT] mini_core_hello_world
/checkout/obj/build/x86_64-unknown-linux-gnu/stage1-tools/cg_gcc/mini_core_hello_world
abc
---
    SUCCESS for generator 'f64 few ones float'. 262144/262144 passed in 5.290861954s
    SUCCESS for generator 'f64 few ones int'. 262144/262144 passed in 5.290830572s
    SUCCESS for generator 'f64 small integer values'. 1048577/1048577 passed in 2.976957493s
    SUCCESS for generator 'f32 small integer values'. 1048577/1048577 passed in 2.316082021s
    SUCCESS for generator 'f64 random digits'. 5000000/5000000 passed in 10800.002767062s
    SUCCESS for generator 'f32 random digits'. 5000000/5000000 passed in 10800.00287117s
    SUCCESS for generator 'f64 large positive exponents'. 10200102/10200102 passed in 395.13159801s
    SUCCESS for generator 'f32 large positive exponents'. 10200102/10200102 passed in 243.157644179s
    SUCCESS for generator 'f64 small exponents'. 12020601/12020601 passed in 10800.002876146s
    SUCCESS for generator 'f32 small exponents'. 12020601/12020601 passed in 121.20577496s
    SUCCESS for generator 'f32 exhaustive'. 4294967296/4294967296 passed in 10800.002837197s
      warning: executed tests != estimated (4294967296 != 4294967295) for exhaustive
    STOPPED for generator 'f64 fuzz'. 2082222021/2082222021 passed in 10800.002874838s
      exited early; exceded 10800s timeout
26/27 tests succeeded in 10805.294163507s (26 passed, 0 failed, 1 stopped)
Command has failed. Rerun with -v to see more details.
  local time: Mon Nov 25 17:26:51 UTC 2024
  network time: Mon, 25 Nov 2024 17:26:51 GMT
##[error]Process completed with exit code 1.
Post job cleanup.
```

</details>

--- a/src/bootstrap/src/core/build_steps/test.rs
+++ b/src/bootstrap/src/core/build_steps/test.rs
@@ -3599,12 +3599,8 @@ impl Step for TestFloatParse {
             &[],
         );
 
-        cargo_run.arg("--");
-        if builder.config.args().is_empty() {
-            // By default, exclude tests that take longer than ~1m.
-            cargo_run.arg("--skip-huge");
-        } else {
-            cargo_run.args(builder.config.args());
+        if !matches!(env::var("FLOAT_PARSE_TESTS_NO_SKIP_HUGE").as_deref(), Ok("1") | Ok("true")) {
+            cargo_run.args(["--", "--skip-huge"]);
         }
 
         cargo_run.into_cmd().run(builder);

